README.md

```diff
@@ -2,327 +2,532 @@
+    // TODO handle response
+  }
+
+  const onError = (error)=> {
+    // TODO handle error
+  }
+
+  visearch.product_search_by_image(parameters, onResponse, onError);
+  ```
+
+- Using image url:
+
+  ```javascript
+  const parameters = {
+    im_url: 'your-image-url'
+  };
+
+  const onResponse = (response)=> {
+    // TODO handle response
+  }
+
+  const onError = (error)=> {
+    // TODO handle error
+  }
+
+  visearch.product_search_by_image(parameters, onResponse, onError);
+  ```
+
+- Using image file:
+
+  ```html
+  <form>
+    Upload image: <input type="file" id="fileUpload" name="fileInput"><br>
+    <input type="submit" value="Submit">
+  </form>
+  ```
+
+  ```javascript
+  const parameters = {
+    image: document.getElementById('fileUpload')
+  };
+
+  const onResponse = (response)=> {
+    // TODO handle response
+  }
+
+  const onError = (error)=> {
+    // TODO handle error
+  }
+
+  visearch.product_search_by_image(parameters, onResponse, onError);
+  ```
+
+> The request parameters for this API can be found [// TODO: link to public documentation]().
+>
+> You will notice that this is the same as some of [ViSearch API](#2-visearch-api) methods. However, on the backend, the logic for search is done differently as mentioned in [ProductSearch API](#3-productsearch-api).
+
+### 3.2 Search by ID
+
+GET /product/search_by_id/{product_id}
+
+Search for visually similar products in the product database giving an indexed product’s unique identifier.
+
+```javascript
+const product_id = 'your-product-id';
+
+const parameters = {
+};
+
+const onResponse = (response)=> {
   // TODO handle response
-}, (err) => {
+}
+
+const onError = (error)=> {
   // TODO handle error
-});
+}
+
+visearch.product_search_by_id(product_id, parameters, onResponse, onError);
 ```
 
+> The request parameters for this API can be found [// TODO: link to public documentation]().
+
 ## 4. Search Results
 
-ViSearch returns a maximum number of 1000 most relevant image search results. You can provide pagination parameters to control the paging of the image search results.
+Javascript does not contain type definitions and the REST API response for all our APIs will instead just convert straight into javascript objects. Here are some of the [ProductSearch API](#3-productsearch-api)'s response object's keys to take note of:
 
-Pagination parameters:
+> For [ViSearch API](#2-visearch-api)'s response, you can find them in the [ViSearch developer doc](https://developers.visenze.com/api/#search-api).
 
 | Name | Type | Description |
-| ---- | ---- | ----------- |
-| page | Integer | Optional parameter to specify the page of results. The first page of result is 1. Defaults to 1. |
-| limit | Integer | Optional parameter to specify the result per page limit. Defaults to 10. |
+|:---|:---|:---|
+| status | String | The request status, either `OK`, `warning`, or `fail` |
+| imId | String | Image ID. Can be used to search again without reuploading |
```

Comment:
this is not correct. You are copying from Java. it should be `im_id`. Please re-check all of the response fields.